### PR TITLE
Fix loading pre-trained DD-PPO checkpoints

### PR DESCRIPTION
**Pull request by rpartsey**
## Motivation and Context

Fixed loading pre-trained DD-PPO PointNav checkpoints (from [DD-PPO README](https://github.com/facebookresearch/habitat-lab/blob/f64ca16901947947cb5a74ede34e5325f2b9f4fc/habitat-baselines/habitat_baselines/rl/ddppo/README.md)). 

- Rolling back visual input normalization done in https://github.com/facebookresearch/habitat-lab/pull/933 fixes loading pre-trained depth and rgb.
- Updating `smash_observation_space` and using `filter_observation_space` in HabGymWrapper fixes loading blind model. 

Added unit test to check if pre-trained DD-PPO PointNav checkpoints can be loaded.

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py [hunk 1]
```diff
@@ -162,27 +162,27 @@ def _setup_actor_critic_agent(self, ppo_cfg: "DictConfig") -> None:
...
             )
-
-        if self.config.habitat_baselines.rl.ddppo.pretrained:
-            self.actor_critic.load_state_dict(
-                {  # type: ignore
-                    k[len("actor_critic.") :]: v
-                    for k, v in pretrained_state["state_dict"].items()
-                }
-            )
-        elif self.config.habitat_baselines.rl.ddppo.pretrained_encoder:
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py:177]: config.habitat_baselines.rl.ddppo.pretrained_encoder is no longer used ?

**Review by vincentpierre — APPROVED** [review]
Just a concern about `pretrained_encoder` config key. It seems the logic is changing.

File: habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py (hunk 1 above)
Comment (rpartsey) [reply habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py:177]: We check `if (self.config.habitat_baselines.rl.ddppo.pretrained_encoder or self.config.habitat_baselines.rl.ddppo.pretrained): ` in line 161.  

So at the point you commented either one of keys will be True or both of them will be True.

Since the `self.config.habitat_baselines.rl.ddppo.pretrained` has "higher priority" we chec it first even if both keys are set to True, else we load the `pretrained_encoder` keys so the logic doesn't change.
